Relaxes the configuration options for influxdb

kirichkov opened the pull request:
**Description:**
Influxdb configuration schema incorrectly set username and password as required options. Default installations of influxdb do not require authentication

**Related issue (if applicable):** fixes #3613

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#1242

**Example entry for `configuration.yaml` (if applicable):**

``` yaml
influxdb:
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [ ] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

By default influxdb allows unauthenticated access
Home Assistant required at least username and password to be present to properly submit data to InfluxDB.
I've removed the completely the test for checking missing keys, as the minimal configuration should work without any additional configuration options

Commit:
Relaxes the configuration options for influxdb

By default influxdb allows unauthenticated access
Home Assistant required at least username and password to be present to properly submit data to influxdb

Commit:
Removes unused import of 'copy'

The copy module was used only in the removed test case responsible for testing the missing keys

fabaff commented on homeassistant/components/influxdb.py:
```diff
@@ -33,8 +35,8 @@
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
```
There are no defaults needed. `username = conf.get(CONF_USERNAME)` and password will be `None` if they are missing in the configuration.

We should also ensure that both (username and password) are present (`vol.Exclusive` / `vol.Inclusive`) if a password is given. As far as I remember is it possible to use only a username without password but I'm not sure.

kirichkov replied:
I just tested it, and in the default InfluxDB configuration, one can supply only username and only password, and it will still work, because Influxdb simply ignores them.

When authentication is enabled, however, one can create username with password that's a blank string, but a blank string is still required to be passed, so it makes sense to require password's presence if username is given.

I will update

Commit:
Updates InfluxDB config schema to require user and password

Current InfluxDB (v 1.0) can work without any authentication, but when authentication is enabled both username and password should be set.

Commit:
Removes extra white space in test_influxdb.py

balloob commented:
🐬  🎉
